sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -661,11 +661,6 @@ def test_check_input_false():
     clf = ElasticNet(selection='cyclic', tol=1e-8)
     # Check that no error is raised if data is provided in the right format
     clf.fit(X, y, check_input=False)
-    X = check_array(X, order='F', dtype='float32')
-    clf.fit(X, y, check_input=True)
```

Comment:
Because it was used for the test below (`assert_raises(ValueError, clf.fit, X, y, check_input=False)`), casting X in 32 bits. But now, _preprocess_data prevent `fit` from raising a `ValueError`, even if `check_input=False`. Since you suggested a smoke test, I can put it back.